core/src/Cuda/Kokkos_Cuda_Parallel_MDRange.hpp

```diff
@@ -95,11 +95,33 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, Kokkos::Cuda> {
 
   inline void execute() const {
     if (m_rp.m_num_tiles == 0) return;
-    const auto maxblocks = m_rp.space().cuda_device_prop().maxGridSize;
+    const auto maxblocks  = m_rp.space().cuda_device_prop().maxGridSize;
+    const auto maxthreads = m_rp.space().cuda_device_prop().maxThreadsDim;
```

Comment:
I meant if you declare maxthreads as dim3 here you won't need to cast (nor compare to zero)

Reply:
dim3 can't be constructed from int[3]

```
    int b[3] = {1,2,3};
    dim3 a = b;
```
```
main.cpp(11): error: no suitable constructor exists to convert from "int [3]" to "dim3"
```